src/interface/guide/components/MajorDefensives/Timeline.tsx

```diff
@@ -102,47 +113,46 @@ const BuffTooltip = ({
   );
 };
 
-export const useMaxMitigationValue = () => {
-  const analyzers = useAnalyzers(MAJOR_ANALYZERS);
-
-  return useMemo(
-    () =>
-      Math.max.apply(
-        null,
-        analyzers?.map((analyzer) =>
-          Math.max.apply(
-            null,
-            analyzer.mitigations.map((mit) => mit.amount),
-          ),
-        ),
-      ),
-    [analyzers],
-  );
-};
-
-const BuffDisplay = ({ hoverKey }: { hoverKey: HoverKey | null }) => {
+const BuffDisplay = <T extends typeof Analyzer>({
+  majorDefensiveAnalyzers,
+  defensiveExpiration,
+  hoverKey,
+  majorDefensives,
+}: {
+  defensiveExpiration: (event: ApplyBuffEvent) => RemoveBuffEvent | undefined;
+  hoverKey: HoverKey | null;
+  majorDefensiveAnalyzers: T[];
+  majorDefensives: MajorDefensiveSpellData[];
+}) => {
   const info = useInfo();
   const events = useEvents();
-  const analyzers = useAnalyzers(TIMELINE_ANALYZERS);
+  const analyzers = useAnalyzers(majorDefensiveAnalyzers);
 
   const tooltipData = useCallback(
     (event: ApplyBuffEvent) => {
-      const analyzer = analyzers?.find((analyzer) => analyzer.appliesToEvent(event));
-      const mit = analyzer?.mitigations.find((mit) => mit.start.timestamp === event.timestamp);
+      const analyzer = analyzers
+        ?.filter(isDefined)
+        ?.find((analyzer) =>
+          isBuffBasedMajorDefensive(analyzer) ? analyzer.appliesToEvent(event) : undefined,
```

Comment:
working through this now. would *highly* discourage having behavior like this conditional on `instanceof`. Not entirely sure we need the split base class either but haven't fully looked at the debuff differences (seems currently incomplete?)

Reply:
I'm likely going to re-go through the implementation of this soon, it's been a bit since I looked at it.

Reply:
okay, then maybe i hold off on porting monk code to it